### PR TITLE
update docker compose with new swift versions

### PR DESCRIPTION
**Pull request by tomerd**
motivation: keep up with the times

changes: add docker compose for 5.8 and 5.9, use stable for 5.7

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

**Review by MaxDesiatov — APPROVED** [review]
LGTM, but maybe new 5.8 and 5.9 configuration could use 22.04 Jammy in addition to or instead of 20.04 Focal?

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Utilities/Docker/docker-compose.2004.58.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+services:
+
+  runtime-setup:
+    image: swift-package-manager:20.04-5.8
+    build:
+      args:
+        ubuntu_version: "focal"
+        swift_version: "5.8"
+        base_image: "swiftlang/swift:nightly-5.8-jammy"
```
[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Utilities/Docker/docker-compose.2004.58.yaml]: Oh, I posted that comment in the middle of your changes, disregard that.

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by tomerd** [comment]
windows failure unrelated

**Pull request merged**